### PR TITLE
feat(sdk): add iterateParamPassStyle and itemPassStyle params to loop

### LINKED ISSUE #1058: Add new sdk support for new pipelineloop syntax
/kind feature

**Description:**
Add new sdk support for new pipelineloop syntax. Two new fields `iterateParamPassStyle` and `itemPassStyle`, both of which can be either `inline` (default value) or `file` will sit on the same spec level as `parallelism`. The initial implementation is sdk support only.

**Additional information:**
[Miscellaneous information that will assist in solving the issue.]

### PR DESCRIPTION
**Pull request by Tomcli**
**Which issue is resolved by this Pull Request:** 
Resolves #1058 

**Description of your changes:**
Add new sdk support for new pipelineloop syntax. Two new fields iterateParamPassStyle and itemPassStyle, both of which can be either inline (default value) or file will sit on the same spec level as parallelism. The initial implementation is sdk support only.

**Environment tested:**

* Python Version (use `python --version`):
* Tekton Version (use `tkn version`):
* Kubernetes Version (use `kubectl version`):
* OS (e.g. from `/etc/os-release`):

**Checklist:**
- [ ] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

**Comment by Tomcli** [comment]
cc @Udiknedormin

**Comment by Tomcli** [comment]
cc @Udiknedormin

**Review by yhwang — APPROVED** [review]

**Comment by Tomcli** [comment]
/hold
need some minor changes

File: sdk/python/kfp_tekton/compiler/compiler.py [hunk 1]
```diff
@@ -577,8 +577,7 @@ def process_parameter(parameter):
       def insert_extra_config_field(config_name, config_object):
         config_value = config_object.lower()
         if config_value in config_value_list:
-          self.loops_pipeline[group_name]['spec'].setdefault('extraConfigs', {})
-          self.loops_pipeline[group_name]['spec']['extraConfigs'][config_name] = config_value
+          self.loops_pipeline[group_name]['spec'][config_name] = config_value
```
Comment (Udiknedormin) [inline sdk/python/kfp_tekton/compiler/compiler.py:580]: It's still defined in the `config_value_list` list, which is non-configurable...

File: sdk/python/kfp_tekton/compiler/compiler.py [hunk 2]
```diff
@@ -572,20 +572,20 @@ def process_parameter(parameter):
           self.loops_pipeline[group_name]['loop_sub_args'] + [sub_group.loop_args.full_name])
       if sub_group.parallelism is not None and sub_group.parallelism > 0:
         self.loops_pipeline[group_name]['spec']['parallelism'] = sub_group.parallelism
-      if hasattr(sub_group, 'iterate_param_pass_style') and sub_group.iterate_param_pass_style is not None:
-        iterate_param_pass_style_value = sub_group.iterate_param_pass_style.lower()
-        if iterate_param_pass_style_value == 'inline' or iterate_param_pass_style_value == 'file':
-          self.loops_pipeline[group_name]['spec']['iterateParamPassStyle'] = iterate_param_pass_style_value
+      config_value_list = ['inline', 'file']
```
[the remark this answers is not in the document]
Comment (Tomcli) [reply sdk/python/kfp_tekton/compiler/compiler.py]: done

File: sdk/python/kfp_tekton/compiler/compiler.py [hunk 3]
```diff
@@ -572,20 +572,20 @@ def process_parameter(parameter):
...
+
+      def insert_extra_config_field(config_name, config_object):
+        config_value = config_object.lower()
+        if config_value in config_value_list:
+          self.loops_pipeline[group_name]['spec'].setdefault('extraConfigs', {})
+          self.loops_pipeline[group_name]['spec']['extraConfigs'][config_name] = config_value
         else:
-          raise ValueError("iterate_param_pass_style value in loop %s must be either 'inline' or 'file', not %s" %
-                           (group_name, iterate_param_pass_style_value))
+          raise ValueError("%s value in loop %s must be either 'inline' or 'file', not %s" %
```
[the remark this answers is not in the document]
Comment (Tomcli) [reply sdk/python/kfp_tekton/compiler/compiler.py]: done

File: sdk/python/kfp_tekton/compiler/compiler.py (hunk 1 above)
Comment (Tomcli) [reply sdk/python/kfp_tekton/compiler/compiler.py]: done

**Review by Udiknedormin — APPROVED** [review]
LGTM

**Review by yhwang — APPROVED** [review]
/lgtm

**Pull request merged**